File: leetcode/editor/en/[392]Is Subsequence.java
392
        Is Subsequence
        2022-11-23 14:17:52

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public boolean isSubsequence(String s, String t) {
        int i = 0;
        int j = 0;
        char[] charOfS = s.toCharArray();
        char[] charOfT = t.toCharArray();

        while (i < charOfS.length && j < charOfT.length) {
            if (charOfS[i] == charOfT[j]) {
                i++;
            }
            j++;
        }

        return i == charOfS.length;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[392]Is Subsequence.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function (s, t) {
    let sIndex = 0;
    let tIndex = 0;

    while (sIndex < s.length && tIndex < t.length) {
        if (s[sIndex] === t[tIndex]) {
            sIndex++;
        }

        tIndex++;
    }

    return sIndex === s.length;
};
//leetcode submit region end(Prohibit modification and deletion)
